src/cascadia/TerminalConnection/ConptyConnection.cpp

```diff
@@ -97,7 +109,11 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
             const auto guidSubStr = std::wstring_view{ wsGuid }.substr(1);
 
             // Ensure every connection has the unique identifier in the environment.
-            environment.emplace(L"WT_SESSION", guidSubStr.data());
+            environment.insert_or_assign(L"WT_SESSION", guidSubStr.data());
+
+            auto wslEnv = environment[L"WSLENV"]; // We always want to load something, even if it's blank.
+            wslEnv = L"WT_SESSION:" + wslEnv; // prepend WT_SESSION to make sure it's visible inside WSL.
```

Comment:
You only need to prepend the key and not also the value? Is `WSLENV` then just a list of keys that WSL should copy from the environment block? That's not totally clear without thinking about it/outside reference so perhaps add that as a comment here.

Reply:
```suggestion
            // WSLENV is a colon-delimited list of environment variables (+flags) that should appear inside WSL
            // https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/
            wslEnv = L"WT_SESSION:" + wslEnv; // prepend WT_SESSION to make sure it's visible inside WSL.
```

Reply:
Yes, OK, perfect. Thank you.